pkg/app/app.go

```diff
@@ -125,6 +126,29 @@ func defaultInputDirectory() configuration.DefaultValueFunction {
 	return callback
 }
 
+func defaultTempDirectory(engine workflow.Engine, logger *zerolog.Logger) configuration.DefaultValueFunction {
+	callback := func(existingValue interface{}) interface{} {
+		if existingValue != nil {
+			return existingValue
+		}
+		ri := engine.GetRuntimeInfo()
+		if ri == nil {
+			return false
+		}
+
+		version := ri.GetVersion()
+		tmpDir := pkg_utils.GetTemporaryDirectory(os.TempDir(), version)
```

Comment:
Issue: The first parameter should be the configured `cache_path`. 

something like this.
```suggestion
		tmpDir := pkg_utils.GetTemporaryDirectory(config.getString(configuration.CACHE_PATH), version)
```